### PR TITLE
Bug 1842177 - Update search engines tests with new unified search UI

### PR DESCRIPTION
**Pull request by sv-ohorvath**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1842177

### CODE REVIEW

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/SearchTest.kt [hunk 1]
```diff
@@ -83,13 +87,37 @@ class SearchTest {
...
+    fun searchBarItemsTest() {
+        navigationToolbar {
+            verifyDefaultSearchEngine("Google")
+            verifySearchBarPlaceholder("Search or enter address")
+        }.clickUrlbar {
+            verifyKeyboardVisibility(true)
+            verifyScanButtonVisibility(true)
+            verifyVoiceSearchButtonVisibility(true)
+            verifySearchBarPlaceholder("Search or enter address")
+            typeSearch("mozillan ")
```
Comment (AaronMT) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/ui/SearchTest.kt:99]: nit: mozillian :)

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/SearchTest.kt [hunk 2]
```diff
@@ -128,29 +156,41 @@ class SearchTest {
         }
     }
 
+    // Verifies temporarily changing the search engine from the Search Shortcut menu
```
Comment (AaronMT) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/ui/SearchTest.kt:159]: nit: // Verifies a temporary change of search engine form the Search shortcut menu

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/SearchTest.kt [hunk 3]
```diff
@@ -83,13 +87,37 @@ class SearchTest {
...
     @Test
-    fun searchScreenItemsTest() {
+    fun searchBarItemsTest() {
+        navigationToolbar {
+            verifyDefaultSearchEngine("Google")
+            verifySearchBarPlaceholder("Search or enter address")
+        }.clickUrlbar {
+            verifyKeyboardVisibility(true)
+            verifyScanButtonVisibility(true)
+            verifyVoiceSearchButtonVisibility(true)
```
Comment (AaronMT) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/ui/SearchTest.kt:97]: I told Andi, I think it's good practice to use named arguments in tests for readability.
Mind applying that throughout here? e.g, 
`verifyScanButtonVisibility(visible = true)`

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/SearchRobot.kt [hunk 4]
```diff
@@ -185,18 +188,17 @@ class SearchRobot {
...
+        searchWrapper().waitForExists(waitingTime)
 
-    fun verifyKeyboardVisibility() = assertKeyboardVisibility(isExpectedToBeVisible = true)
-    fun verifySearchEngineList(rule: ComposeTestRule) = rule.assertSearchEngineList()
+        assertEquals(
+            "Keyboard not shown",
+            isExpectedToBeVisible,
+            mDevice
+                .executeShellCommand("dumpsys input_method | grep mInputShown")
+                .contains("mInputShown=true"),
```
[the remark this answers is not in the document]
Comment (sv-ohorvath) [reply fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/SearchRobot.kt:199]: Done, but it added a few other files to the changelog.

**Review by AaronMT — APPROVED** [review]

**Pull request merged**